src/main/java/com/hms_networks/americas/sc/extensions/connectors/framework/AbstractConnectorConfig.java

```diff
@@ -538,12 +538,32 @@ public JSONObject getDefaultConfigurationObject() throws JSONException {
    * @since 1.15.2
    */
   protected void logMissingField(String missingKey, String defaultValue) {
-    Logger.LOG_WARN(
+    final boolean suppressAdditional = false;
+    logMissingField(missingKey, defaultValue, suppressAdditional);
+  }
+
+  /**
+   * Logs a missing field in the configuration file along with the default value that will be used.
+   * If the specified {@code suppressAdditional} boolean is {@code true}, the event will only be
+   * logged once (tracked by the key).
+   *
+   * @param missingKey the JSON key of the field that was missing/could not be found
+   * @param defaultValue the default value that will be used
+   * @param logOnce whether to log only once and suppress additional log events for the same key
+   * @since 1.15.9
+   */
+  protected void logMissingField(String missingKey, String defaultValue, boolean logOnce) {
+    String logMessage =
         "The "
             + missingKey
             + " option was not found in the configuration file. Using default value of "
             + defaultValue
-            + ".");
+            + ".";
+    if (logOnce) {
+      Logger.LOG_WARN_ONCE(missingKey, logMessage);
```

Comment:
this key implementation is not very unique. If another log message was added referencing the same config file key and used the same scheme it would not be printed but it would be unique.

Reply:
This method is within the abstract connector framework's configuration class, and the goal of the `.logMissingField` methods is to provide a single point for handling log output associated with missing configuration fields. 

I opted for directly using the key here because it is generally safe to assume there is no other need for another/other log messages elsewhere.

Reply:
I think even a simple improvement would be 

`Logger.LOG_WARN_ONCE("logMissingField:"+missingKey, logMessage);`

Its not unreasonable to thing someone may add in somewhere else
`Logger.LOG_WARN_ONCE(invalidValueKey, logMessage);`
and it would not be printed because the key could be exactly the same.

Reply:
Added a uniqueness prefix